src/Build/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -110,14 +110,14 @@ protected void DecorateItemsWithMetadata(ICollection<I> items, ImmutableList<Par
 
                     // Do not expand properties as they have been already expanded by the lazy evaluator upon item operation construction.
                     // Prior to lazy evaluation ExpanderOptions.ExpandAll was used.
-                    const ExpanderOptions metadataExpansionOptions = ExpanderOptions.ExpandMetadata | ExpanderOptions.ExpandItems;
```

Comment:
Thanks for the all the digging+explanation! It makes sense now :) It would be useful to have this in the commit message with the tests that you will be adding. 👍